NAS2D/Renderer/Fade.cpp

```diff
@@ -65,7 +65,7 @@ void Fade::update()
 		return;
 	}
 
-	const auto step = static_cast<uint8_t>(std::clamp(mFadeTimer.accumulator() * 255u / mDuration, 0u, 255u));
+	const auto step = static_cast<uint8_t>(std::clamp(mFadeTimer.accumulator() * 255u / static_cast<unsigned int>(mDuration.count()), 0u, 255u));
```

Comment:
I feel like there should be a way to avoid this cast here. Particularly since the returned `mFadeTimer.accumulator()` value should be effectively the same type as `mDuration.count()`. Effectively we want to use the natural type of the expression.

Reply:
The default type of `milliseconds` is `long long`, the cast was just me being paranoid with a potentially unsigned/signed mismatch.

Reply:
Nope. Cast is required. Two errors pop up:

`A sub-expression may overflow before being assigned to a wider type` for the division and `no matching overload found for std::clamp arguments are long long, unsigned int, unsigned int`

Reply:
There may be other ways to address and avoid these problems. A `static_cast` might not be the only solution. Another consideration might be to change the `u` type suffix on the constants. That may not be so appropriate now if the other associated types have changed by the transition to `std::chrono::millisecond`.

It may also be possible to use an explicit type parameter when calling `std::clamp`.

Reply:
After looking more closely, `mFadeTimer.accumulator()` returns an `unsigned int`, and is based on [`SDL_GetTicks`](https://wiki.libsdl.org/SDL_GetTicks), which return `Uint32`. Meanwhile `mDuration.count()` is calling [`std::chrono::millisecond::count()`](https://en.cppreference.com/w/cpp/chrono/duration/count), where for type [`std::chrono::millisecond`](https://en.cppreference.com/w/cpp/chrono/duration) the return type `Rep` is `/*signed integer type of at least 45 bits*/`.

Since we are using two separate types, there will be a `static_cast` here.

----

This does make me wonder if we should really be using a consistent type, all relating to a single source of ticks. Though that's perhaps a separate issue.